### PR TITLE
Handle the case where the Marshal.ThrowExceptionForHR is inlined in

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
the test assembly and the Source isn't corlib.

cc @stephentoub @jkoritzinsky

### CODE REVIEW

**Comment by AaronRobinsonMSFT** [comment]
Fixes https://github.com/dotnet/coreclr/issues/22244

File: src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/ThrowExceptionForHRTests.cs [hunk 1]
```diff
@@ -30,14 +31,13 @@ public void ThrowExceptionForHR_NoErrorInfo_ReturnsValidException(int errorCode)
...
-                if (PlatformDetection.IsNetCore)
-                {
-                    Assert.Equal("System.Private.CoreLib", ex.Source);
-                }
-                else
-                {
-                    Assert.Equal("mscorlib", ex.Source);
-                }
+
+                string sourceMaybe = PlatformDetection.IsNetCore
```
Comment (stephentoub) [inline src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/ThrowExceptionForHRTests.cs]: Is this going to work correctly on .NET Native? Mono?

Comment (stephentoub) [reply src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/ThrowExceptionForHRTests.cs]: Though I guess this was the logic we already had?

Comment (AaronRobinsonMSFT) [reply src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/ThrowExceptionForHRTests.cs]: That is a great question. I have no idea. I assumed the logic of choice was correct. How would/could I verify this?

Comment (AaronRobinsonMSFT) [reply src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/ThrowExceptionForHRTests.cs]: > Though I guess this was the logic we already had?

Yep. I just added the option of the result being the current assembly. I had hoped this was already passing on other runtimes.

**Pull request merged**